nailgun/entities.py

```diff
@@ -1970,27 +1983,14 @@ def create_missing(self):
         super(LifecycleEnvironment, self).create_missing()
         if (self.name != 'Library' and  # pylint:disable=no-member
                 not hasattr(self, 'prior')):
-            response = client.get(
-                self.path('base'),
-                data={
-                    u'name': u'Library',
-                    # pylint:disable=no-member
-                    u'organization_id': self.organization.id,
-                },
-                **self._server_config.get_client_kwargs()
-            )
-            response.raise_for_status()
-            results = response.json()['results']
+            results = self.search({'organization'}, {u'name': u'Library'})
```

Comment:
> I just don't understand difference between `{'organization'}` and `{u'name': u'Library'}` parameters representation

The first argument specifies which fields nailgun should use when automatically generating a search expression for you. The second argument is merged in to the first - it allows for customization after a search has been generated. Here's a simple example of how the queries are merged:

```python
>>> from nailgun import entities
>>> org = entities.Organization(name='foo', description='bar')                                            
>>> org.search_payload()
{'name': 'foo', 'description': 'bar'}
>>> org.search_payload({'name'})
{'name': 'foo'}
>>> org.search_payload({'name'}, {'per_page': 50})
{'per_page': 50, 'name': 'foo'}
```

I will fully admit that the search generator currently *sucks*. It's awfully naive. But I'm OK with that - I know that we have the tools in hand to make it better, and I want us to put some effort in to doing so rather than writing manual search queries for the rest of our days.